vars/skipStage.groovy

```diff
@@ -156,12 +156,19 @@ boolean skip_ftest_valgrind(String distro, String target_branch) {
 }
 
 boolean skip_ftest_hw(String size, String target_branch) {
-    return already_passed() ||
-           env.DAOS_STACK_CI_HARDWARE_SKIP == 'true' ||
-           !paramsValue('CI_' + size.replace('-', '_') + '_TEST', true) ||
-           skip_stage_pragma('func-test') ||
+    if (already_passed() || !testsInStage()) {
+        // Stage has already run or there are no tests to run in the stage - always skip
+        return true
+    }
+    if (paramsValue('CI_' + size.replace('-', '_') + '_TEST', false) ||
+        !skip_stage_pragma('func-hw-test-' + size, true) ||
```

Comment:
Second argument to `skip_stage_pragma()` is a `String`:
https://github.com/daos-stack/pipeline-lib/blob/79760290f865a55896cdc9a2dee0218537db4f27/vars/skipStage.groovy#L85
and you are passing a Boolean to it here.

```suggestion
        !skip_stage_pragma('func-hw-test-' + size, 'true') ||
```

Indeed, a variant of this function that takes a `Boolean` and returns a `Boolean` so that it can be used in logic statements would indeed be useful.  It has occurred to me many times but never been itchy enough to spend the time writing it.

Reply:
Fixed, thanks.